File: imate_android_app/app/src/main/java/com/inty/imate/chat/data/ChatTextSendRequestFactory.kt
package com.inty.imate.chat.data

import com.inty.imate.chat.data.bean.SendMsgReq
import com.inty.imate.chat.data.bean.SendMsgReqMessage
import com.inty.imate.chat.data.datasource.ChatWebSocketRemoteDataSource
import java.util.UUID
import kotlinx.serialization.json.JsonPrimitive

object ChatTextSendRequestFactory {

    fun buildTextSendMsgReq(agentId: String, userText: String): SendMsgReq {
        val trimmed = userText.trimEnd()
        return SendMsgReq(
            messages =
                listOf(
                    SendMsgReqMessage(
                        role = "user",
                        content = JsonPrimitive(trimmed),
                    )
                ),
            timeContext = ChatWebSocketRemoteDataSource.buildUserTimeContextOrNull(),
            targetImateId = agentId,
            messageId = UUID.randomUUID().toString(),
        )
    }
}

File: imate_ios_app/imate/chat/ChatTextSendRequestFactory.swift
import Foundation

enum ChatTextSendRequestFactory {
    static func buildTextSendMsgReq(agentId: String, userText: String) -> SendMsgReq {
        var trimmed = userText
        while let last = trimmed.unicodeScalars.last,
              CharacterSet.whitespacesAndNewlines.contains(last)
        {
            trimmed.removeLast()
        }
        return SendMsgReq(
            messages: [SendMsgReqMessage(role: "user", content: trimmed)],
            model: "chatbot",
            stream: false,
            timeContext: UserTimeContextBuilder.buildNow(),
            targetImateId: agentId,
            messageId: UUID().uuidString,
            messageType: CompanionChatTurnMessageType.userMessage
        )
    }
}
